### PR TITLE
BA-1721: Add Component Testing to the @baseapp-frontend Pipeline

### PR DESCRIPTION
**Pull request by luisguareschi**
Acceptance Criteria 
To ensure that both unit and component tests are executed as part of the pipeline, we will be updating the current setup to accommodate component testing. This involves renaming the existing test step to reflect that it only covers unit tests and adding a new step for component tests executed by Cypress.

Rename the existing "Test" step in GitHub Actions to "Unit Test" to clarify that it is only responsible for unit tests.

Update the existing unit test script on the `@baseapp-frontend` packages (utils and authentication) from `test` to `test:unit` to reflect its specificity.

Add a new npm script `"test:component"` in the `@baseapp-frontend/components` package.  

Ensure this script runs Cypress to perform headless component testing on all tests cases available inside the package

In the main GitHub Actions Main workflow, add both the `test:unit` and `test:component` scripts to the pipeline.

Ensure that component tests are run as a separate step following the unit tests. 

Notes 

Existent component test we have under @baseapp-frontend/components: https://github.com/silverlogic/baseapp-frontend/blob/master/packages/components/modules/comments/Comments/__tests__/Comments.cy.tsx 

Approvd 
https://app.approvd.io/silverlogic/BA/stories/36005

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -75,4 +75,27 @@ jobs:
         run: pnpm install --frozen-lockfile
 
       - name: Test Applications and Packages
```
Comment (anicioalexandre) [inline .github/workflows/main.yml:77]: let's change to `Unit Test Applications and Packages` and the component test to `Component Test Applications and Packages`

**Review by anicioalexandre — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (luisguareschi) [reply .github/workflows/main.yml:77]: @anicioalexandre Done!

**Comment by priscilladeroode** [comment]
@luisguareschi and @anicioalexandre I think we need to generate a new version of the package, maybe a patch increase?

**Comment by anicioalexandre** [comment]
> @luisguareschi and @anicioalexandre I think we need to generate a new version of the package, maybe a patch increase?

@priscilladeroode in that case, I dont think we need new versions, since most of the work was changing pipelines and altering test scripts, it won't really affect the packages code

**Review by priscilladeroode — APPROVED** [review]

**Pull request merged**